WalletWasabi.Fluent/Models/Wallets/WalletLoadWorkflow.cs

```diff
@@ -58,16 +59,9 @@ public void Start()
 							 .DisposeWith(_disposables);
 
 		Observable.FromEventPattern<bool>(Services.Synchronizer, nameof(Services.Synchronizer.ResponseArrivedIsGenSocksServFail))
```

Comment:
Out of the scope of this PR, but we can discuss it for a following up one.

##

```c#
Services.Synchronizer.WhenAnyValue(x => x.BackendStatus)
  .SubscribeAsync(async status => await LoadWalletAsync(isBackendAvailable: status == BackendStatus.Connected).ConfigureAwait(false))
  .DisposeWith(_disposables);
```

The problem with this code is if you login to a wallet before the backend is connected, `LoadWalletAsync()` method would be immediately called thinking the backend is not connected. Because when you execute `SubscribeAsync()` it runs on subscription.
You could use `Skip(1)` so it won't run on subscription and would wait until the backend connects, but it wouldn't work in the case of the backend being down. Because the value of `BackendStatus` would never change. That is why you need the event `ResponseArrivedIsGenSocksServFail` for detecting that.

##

> Then there are two competing pieces of code:

Yes, it is true.

> Seems like a big hammer or is it a workaround for some missing reactive API? (Like "HandleOnlyFirstChange"?)

Yes, good point.

We could do (not tested):
```c#
var backendError = Observable
	.FromEventPattern(Services.Synchronizer, nameof(Services.Synchronizer.ResponseArrivedIsGenSocksServFail))
	.Select(_ => Services.Synchronizer.BackendStatus == BackendStatus.Connected)
	.Where(x => !x);

var backendConnected = Services.Synchronizer
	.WhenAnyValue(x => x.BackendStatus)
	.Select(x => x == BackendStatus.Connected)
	.Where(x => x);

Observable
	.Merge(backendError, backendConnected)
	.Take(1)
	.SubscribeAsync(async isBackendAvailable => await LoadWalletAsync(isBackendAvailable).ConfigureAwait(false))
	.DisposeWith(_disposables);
```

It ensures concurrency and also unsubscribes right after it ran at least once.

Reply:
cACK